### PR TITLE
Save Objects::hash varargs array allocation on JarResource::hashCode

### PR DESCRIPTION
**Pull request by franz1981**
Starting up  quarkus app in JVM mode still allocate useless varargs Object[] while computing JarResource::hashCode, see the allocation flamegraphs at:

image

The code there is unlikely to become hot and avoid performing such allocation...

### CODE REVIEW

**Review by geoand — APPROVED** [review]

File: independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/JarResource.java [hunk 1]
```diff
@@ -185,6 +185,8 @@ public boolean equals(Object o) {
 
     @Override
     public int hashCode() {
-        return Objects.hash(manifestInfo, jarPath);
+        int result = Objects.hashCode(manifestInfo);
+        result = 31 * result + Objects.hashCode(jarPath);
+        return result;
```
Comment (gsmet) [inline independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/JarResource.java:190]: Ah yeah, we banned `Objects.hash` from Hibernate Validator for this reason.

I think we can even go further and remove the `manifestInfo` from the `equals()` and `hashCode()`: the unicity should be done on the `Path` really, we don't care about what's in the `Manifest`, we will never have a jar with the same `Path` and a different `ManifestInfo`.

@aloubyansky could you confirm ^?

Comment (aloubyansky) [reply independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/JarResource.java:190]: Right, sounds good. Thanks!

Comment (gsmet) [reply independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/JarResource.java:190]: @franz1981 can you adjust or do you want me to take care of it?

Comment (franz1981) [reply independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/JarResource.java:190]: If you could (and put you as co-author if you wish) would be great...I'm still catching up with the email :(

Comment (gsmet) [reply independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/JarResource.java:190]: Sure will do, didn't want to step on your toes :). Will add a separate commit as I think we need to keep the semantic of both changes.

**Review by gsmet — APPROVED** [review]

**Pull request merged**